src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,120 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    static final int DEFAULT_INITIAL_CAPACITY = 16;
+    static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int capacity;
+    private int threshold;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+        threshold = (int) (DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+        capacity = DEFAULT_INITIAL_CAPACITY;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size + 1 > threshold) {
+            resize();
+        }
+        putNode(key, value);
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        int hash = getNumberCellByKey(key);
+        Node<K, V> currentNode = table[hash];
+        if (currentNode != null) {
+            while (currentNode.next != null) {
+                if (isEqualsKeys(currentNode.key, key)) {
+                    return currentNode.value;
+                } else {
+                    currentNode = currentNode.next;
+                }
+            }
+            return currentNode.value;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void resize() {
+        capacity = capacity * 2;
+        threshold = (int) (capacity * DEFAULT_LOAD_FACTOR);
+        Node<K, V>[] oldTable = table;
+        table = new Node[capacity];
+        copyElement(oldTable);
+    }
+
+    private void copyElement(Node<K, V>[] oldTable) {
+        for (Node<K, V> element : oldTable) {
+            if (element != null) {
+                Node<K, V> currentNode = element;
+                if (currentNode.next == null) {
+                    putNode(currentNode.key, currentNode.value);
+                } else {
+                    while (currentNode != null) {
+                        putNode(currentNode.key, currentNode.value);
+                        currentNode = currentNode.next;
+                    }
+                }
+
+            }
+        }
```

Comment:
```suggestion
        for (Node<K, V> currentNode : oldTable) {
            while (currentNode != null) {
                putNode(currentNode.key, currentNode.value);
                currentNode = currentNode.next;
            }
        }
```
much simpler way to iterate through table